internal/operator-controller/applier/loader.go

```diff
@@ -0,0 +1,59 @@
+package applier
+
+import (
+	"fmt"
+	"io/fs"
+
+	"helm.sh/helm/v3/pkg/chart"
+
+	"github.com/operator-framework/operator-controller/internal/operator-controller/rukpak/bundle/source"
+	imageutil "github.com/operator-framework/operator-controller/internal/shared/util/image"
+)
+
+const (
+	BundleTypeRegistryV1 BundleType = "registry-v1"
+	BundleTypeHelm       BundleType = "helm"
+)
+
+type BundleType string
+
+type HelmChartLoaderMap map[BundleType]HelmChartLoader
+
+type BundleFSChartLoader struct {
+	HelmChartLoaders HelmChartLoaderMap
+}
+
+func (b BundleFSChartLoader) Load(bundleFS fs.FS, installNamespace string, watchNamespace string) (*chart.Chart, error) {
+	bundleType := b.determineBundleType(bundleFS)
+	if loader, ok := b.HelmChartLoaders[bundleType]; ok {
+		return loader.Load(bundleFS, installNamespace, watchNamespace)
+	}
+	return nil, fmt.Errorf("unsupported bundle type: %s", bundleType)
+}
+
+// determineBundleType is a stand-in method to helm determine the bundle type until
+// the bundle metadata can be given together with the FS to the Load method
+func (b BundleFSChartLoader) determineBundleType(bundleFS fs.FS) BundleType {
+	if imageutil.IsBundleSourceChart(bundleFS) {
+		return BundleTypeHelm
+	}
```

Comment:
(I know this PR is scoped to refactoring how the Helm applier gets it chart, so this is more of a follow-up we should cover....)

I don't think it makes sense for the `shared/util/image` package to:
1. Contain helpers that have nothing to do with OCI images (this helper reads a bundleFS to decide if it contains a chart or not; that's not related to images)
2. Contain anything related to helm (or any other bundle format _at all_). I imagine that this library should continue evolving to be the generic framework for turning an OCI ref into an `fs.FS`, where the actual implementation of that functionality is provided _to_ the library, not _part of_ the library.